debian/wview-mysql-export.1
.\"                                      Hey, EMACS: -*- nroff -*-
.\" First parameter, NAME, should be all caps
.\" Second parameter, SECTION, should be 1-8, maybe w/ subsection
.\" other parameters are allowed: see man(7), man(1)
.TH WVIEW-MYSQL-EXPORT 1 "November 19, 2009"
.\" Please adjust this date whenever revising the manpage.
.\"
.\" Some roff macros, for reference:
.\" .nh        disable hyphenation
.\" .hy        enable hyphenation
.\" .ad l      left justify
.\" .ad b      justify to both left and right margins
.\" .nf        disable filling
.\" .fi        enable filling
.\" .br        insert line break
.\" .sp <n>    insert n+1 empty lines
.\" for manpage-specific macros, see man(7)
.SH NAME
wview-mysql-export \- script to export wview SQLite3 data to a MySQL Database on local or remote server
.SH SYNOPSIS
.B wview-mysql-export
.RI
.br
.SH DESCRIPTION
This manual page describes the use of
.B wview-mysql-export
.
Export wview weather data (archive and HILOW) to a local or remote MySQL server.
.BR
.BR Note: REQUIRES: mysqlimport CLI utility for database export.
.BR Note: must be executed on the source wview server.
.BR Note: Usually executed as a cron job.
.SH OPTIONS
There are no options.
.TP
.SH SEE ALSO
.BR wview-mysql-create (1),
.BR /usr/share/doc/wview/wview-User-Manual.html
.br
.SH AUTHOR
wview-mysql-export was written by Mark Teel.
.PP
This manual page was written by Mark Teel <mteel2005@gmail.com>,
for the Debian project (and may be used by others).


ChangeLog
RELEASE VERSION HISTORY (newest to oldest)
------------------------------------------


5.21.5    04-06-2014
--------------------

radlib version required: 2.12.0 or newer

1)  Remove debug message.




5.21.4    04-06-2014
--------------------

radlib version required: 2.12.0 or newer

1)  Added stale pid file cleanup to the APT package start script.

2)  WMRUSB HID interface changes. Changed reader thread to blocking IO.
    Mitigated the console occasionally blasting out bogus data. Now zero
    packet loss. Fixed issue with pressure sanity check. Added sanity
    check for bogus rain accumulator values from the WMR console.




5.21.3    03-31-2014
--------------------

radlib version required: 2.12.0 or newer

1)  Further WMRUSB interface improvements. Better packet framing.




5.21.2    03-30-2014
--------------------

radlib version required: 2.12.0 or newer

1)  Updated debian APT postinst script to include copying the exfoliation skin
    for new installs.

2)  Enabled checksum processing for WMR stations.

3)  Changed LOOP windspeed items to floats for better precision.
    NOTE: If you use custom datafeed clients the LOOP data structure has changed.

4)  Added temp only packet support for WMRUSB stations. This adds support for
    add-on sensors such as the pool sensor.

5)  Fixed APT installs changing the wview.log file ownership incorrectly. Also
    restarts rsyslog automatically if a new install.




5.21.1    03-13-2014
--------------------

radlib version required: 2.12.0 or newer

1)  Modified HILOW database storage so higher precision is used for cumulative
    data such as rain and ET.

2)  Fixed issue with yearly historical charts when generated at wview start
    (as opposed to at midnight). The current day was erroneously being included in
    the history data. This was most apparent with the UV and Solar Radiation
    charts.

3)  Fixed bug introduced in 5.21.0 release causing NULL wind direction to have
    360 added to it (to normalize calibration result). A value of -99640 was
    being stored/displayed.




5.21.0    03-05-2014
--------------------

radlib version required: 2.12.0 or newer

1)  WMRUSB improvements. Further reverse engineering of the undocumented protocol. Reset and
    heartbeat messages now working. Packet reception more stable. Initial packet acquisition
    faster resulting in faster startup.

2)  Modify CWOP packet header for new stricter compliance requirements of CWOP servers.

3)  Fix log message time_t bug for 64-bit platforms.

4)  Add timeout parameter for ssh rules. Added rows to config database.

5)  Updated data feed conversion logic to handle ARCHIVE_VALUE_NULL in loop data.

6)  Added flat line sensor value detection for outside temp, wind direction and wind speed.
    If there are no changes over a 4 hour period an email alert will be sent.

7)  Added Station Name, City and State to the wviewmgmt System Status page header.

8)  Added logic to SSH timer to fixup next timer interval if system time has skewed or
    if the ssh transfer took longer than 2 minutes.

9)  Fixed TE923 rain calculation to assume each tip is 1/36" instead of 1mm.

10) Fixed a bug for Vantage Pro stations which could cause the last archive interval of
    the previous day to be included in the current day HILOW data. This occurred if the
    midnight archive record was provided late by the console.

11) Added wet bulb temperature calculation and html tag --WetBulbTemp--. Added to all
    "readings.incx" templates.

12) Added email alert notification if no loop data has been received for a given archive
    interval. This applies to all stations except the Davis stations (alerts already
    exist for them).

13) Updated start scripts to detect and delete stale pid files.

14) Modified WH1080 data parsing to assume a humidity of 9% if the value from the
    station is undefined (0xff).




5.20.2    03-21-2012
--------------------

radlib version required: 2.12.0 or newer

1)  Added back "dailyrainin" for http reporting. They do no computation based on hourly rain values.

2)  Fixed a typo in the wviewmgmt station.php file. 




5.20.1    03-20-2012
--------------------

radlib version required: 2.12.0 or newer

1)  Modified http rain submission (Wunderground and Weatherforyou) so only the
    "rainin" parameter is sent so daily rain will be correct. 




5.20.0    03-17-2012
--------------------

radlib version required: 2.12.0 or newer

1)  Added "wait_for_time_set" function to most wview start script examples so 
    wview daemons are not started until the system time is set. This applies to
    platforms without a real time clock which must reset their time during 
    startup.

2)  Updated the User Manual section on skin selection to better describe enabling
    extended data generation (UV, ET, radiation).

3)  Updated the debian post install script to try harder to find the default http
    document root location for creation of symbolic links to wviewmgmt and the
    weather home page.

4)  Added new configuration option to optionally disable using the Vantage Pro 
    console archive records and instead autogenerate them from the LOOP data.
    This is how all other stations work. It also supports non-standard console
    interfaces (those not using the datalogger on the console).

5)  Updated the Weatherforyou URL for data submissions.

6)  Fixed bug with DataFeed clients that caused an infinite loop if you
    restarted wview while there was a datafeed client connection.

7)  Fixed bug with Alarm Type configuration introduced in 5.19.0.

8)  Changed TWI station to compute rain fall rate as it is reported to not
    be provided by the station.

9)  Added socket retry logic to the station ethernet interface (used by the
    Vantage Pro IP datalogger and any serial-to-ethernet device).

10) Modified CWOP host name to IP address resolution so that 'getaddrinfo' is
    used instead of 'gethostbyname' and all results are tried in the order they 

dbexport/wview-mysql-export.sh
################################################################################
#
# File:           wview-mysql-export.sh
#
# Description:    Provide a script to export SQLite3 wview data to a MySQL 
#                 database. Should be run periodically via cron.
#
# Usage:          wview-mysql-export
#                 Arguments:
#                   debug        - indicate to log verbosely and to
#                                  create the MySQL debug log file 
#                                  $prefix/etc/wview/export/mysql_import.log
#                   create       - create MySQL user, database and tables based
#                                  on config values in 
#                                  $prefix/etc/wview/wview-conf.sdb, then exit
#                                  (should not be given for cron execution)
#
# History:
# Engineer	  Date	    Ver   Comments
# MS Teel	  03/28/09   1    Original
#
# Notes:          The marker file approach is inspired by Jerry Fiddler.
#                 Requires: sqlite3 CLI utility, mysqlimport CLI utility and
#                           mysql CLI utility for database creation.
#
################################################################################

################################################################################
#################################  M A C R O S  ################################
################################################################################

## DO NOT CHANGE ANY DEFS IN THIS BLOCK!

export PATH=/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin

SECONDS_IN_DAY=86400
SECONDS_IN_HOUR=3600

ARCHIVE_DB_FILE=$WVIEW_DATA_DIR/archive/wview-archive.sdb
HILOW_DB_FILE=$WVIEW_DATA_DIR/archive/wview-hilow.sdb
NOAA_DB_FILE=$WVIEW_DATA_DIR/archive/wview-noaa.sdb

## The defaults match the SQLite table names: 
SQL_ARCHIVE_TABLE=archive
SQL_HILOW_TABLE_INTEMP=inTemp
SQL_HILOW_TABLE_OUTTEMP=outTemp
SQL_HILOW_TABLE_INHUMIDITY=inHumidity
SQL_HILOW_TABLE_OUTHUMIDITY=outHumidity
SQL_HILOW_TABLE_BP=baromPressure
SQL_HILOW_TABLE_WINDSPEED=windSpeed
SQL_HILOW_TABLE_WINDGUST=windGust
SQL_HILOW_TABLE_DEWPOINT=dewPoint
SQL_HILOW_TABLE_RAIN=rain
SQL_HILOW_TABLE_RAINRATE=rainRate
SQL_HILOW_TABLE_WINDCHILL=windChill
SQL_HILOW_TABLE_HEATINDEX=heatIndex
SQL_HILOW_TABLE_ET=ET
SQL_HILOW_TABLE_UV=UV
SQL_HILOW_TABLE_SOLARRAD=solarRadiation
SQL_HILOW_TABLE_HAIL=hail
SQL_HILOW_TABLE_HAILRATE=hailRate
SQL_HILOW_TABLE_WINDDIR=windDir
SQL_NOAA_TABLE=noaaHistory

TIME_MARKER_FILE=$WVIEW_CONF_DIR/export/mysql_export_marker
ARCHIVE_MARKER_FILE=$WVIEW_CONF_DIR/export/archive_marker
HILOW_MARKER_FILE=$WVIEW_CONF_DIR/export/hilow_marker
NOAA_MARKER_FILE=$WVIEW_CONF_DIR/export/noaa_marker
ARCHIVE_EXPORT_FILE=$WVIEW_CONF_DIR/export/$SQL_ARCHIVE_TABLE.csv
HILOW_EXPORT_FILE_INTEMP=$WVIEW_CONF_DIR/export/$SQL_HILOW_TABLE_INTEMP.csv
HILOW_EXPORT_FILE_OUTTEMP=$WVIEW_CONF_DIR/export/$SQL_HILOW_TABLE_OUTTEMP.csv
HILOW_EXPORT_FILE_INHUMIDITY=$WVIEW_CONF_DIR/export/$SQL_HILOW_TABLE_INHUMIDITY.csv
HILOW_EXPORT_FILE_OUTHUMIDITY=$WVIEW_CONF_DIR/export/$SQL_HILOW_TABLE_OUTHUMIDITY.csv
HILOW_EXPORT_FILE_BP=$WVIEW_CONF_DIR/export/$SQL_HILOW_TABLE_BP.csv
HILOW_EXPORT_FILE_WINDSPEED=$WVIEW_CONF_DIR/export/$SQL_HILOW_TABLE_WINDSPEED.csv
HILOW_EXPORT_FILE_WINDGUST=$WVIEW_CONF_DIR/export/$SQL_HILOW_TABLE_WINDGUST.csv
HILOW_EXPORT_FILE_DEWPOINT=$WVIEW_CONF_DIR/export/$SQL_HILOW_TABLE_DEWPOINT.csv
HILOW_EXPORT_FILE_RAIN=$WVIEW_CONF_DIR/export/$SQL_HILOW_TABLE_RAIN.csv
HILOW_EXPORT_FILE_RAINRATE=$WVIEW_CONF_DIR/export/$SQL_HILOW_TABLE_RAINRATE.csv
HILOW_EXPORT_FILE_WINDCHILL=$WVIEW_CONF_DIR/export/$SQL_HILOW_TABLE_WINDCHILL.csv
HILOW_EXPORT_FILE_HEATINDEX=$WVIEW_CONF_DIR/export/$SQL_HILOW_TABLE_HEATINDEX.csv
HILOW_EXPORT_FILE_ET=$WVIEW_CONF_DIR/export/$SQL_HILOW_TABLE_ET.csv
HILOW_EXPORT_FILE_UV=$WVIEW_CONF_DIR/export/$SQL_HILOW_TABLE_UV.csv
HILOW_EXPORT_FILE_SOLARRAD=$WVIEW_CONF_DIR/export/$SQL_HILOW_TABLE_SOLARRAD.csv
HILOW_EXPORT_FILE_HAIL=$WVIEW_CONF_DIR/export/$SQL_HILOW_TABLE_HAIL.csv
HILOW_EXPORT_FILE_HAILRATE=$WVIEW_CONF_DIR/export/$SQL_HILOW_TABLE_HAILRATE.csv
HILOW_EXPORT_FILE_WINDDIR=$WVIEW_CONF_DIR/export/$SQL_HILOW_TABLE_WINDDIR.csv
NOAA_EXPORT_FILE=$WVIEW_CONF_DIR/export/$SQL_NOAA_TABLE.csv

ARCHIVE_ENABLE=1
HILOW_ENABLE=1
NOAA_ENABLE=1
UPDATE_MARKER=0

MYSQL_IMPORT_LOG_FILE=$WVIEW_CONF_DIR/export/mysql_import.log

## Test command line arguments:
DO_CREATION=0
if [ "$1" = "help" ]; then
    echo "USAGE:"
    echo "wview-mysql-export [help | debug | create]"
    echo "    <no arguments>                           - (default behavior) export SQLite3 data to MySQL, typically executed as a cron job"
    echo ""
    echo "    help                                     - print out usage directions and exit"
    echo ""
    echo "    debug                                    - log verbosely and create the MySQL debug log file $WVIEW_CONF_DIR/export/mysql_import.log"
    echo ""
    echo "    create <mysql_root_passwd>               - create local MySQL user, database and tables based on config values in" 
    echo "                                               $WVIEW_CONF_DIR/wview-conf.sdb, then exit (no exports done);"
    echo "                                               (should not be given for cron execution)"
    echo ""
    echo ""
    echo "REQUIRES: sqlite3 CLI utility, mysqlimport CLI utility and mysql CLI utility for database creation."
    echo ""
    exit 0
fi

if [ "$1" = "debug" ]; then
    VERBOSE=1
else
    VERBOSE=0
fi

if [ "$1" = "create" ]; then
    if [ x"$2" = x ]; then
        echo "You must specify the MySQL root user password to create local export database - use 'wview-mysql-export help' for details"
        exit 1
    else
        MYSQL_ROOT_PASSWORD=$2
        DO_CREATION=1
    fi
fi

################################################################################
#######################  D E F I N E  F U N C T I O N S  #######################
################################################################################
syslog()
{
	if [ $VERBOSE != 0 ]; then
        logger -t WV_EXPORT $1 $2
    fi
}

syslog_error()
{
    logger -t WV_EXPORT_ERROR $1 $2
}

# $1 = DB file (full path)
# $2 = table name
# $3 = dateTime stamp
# $4 = CSV output file
sqlite3_export()
{
    echo ".mode list" > /tmp/commands.sql
    echo ".separator ," >> /tmp/commands.sql
    echo ".nullvalue '\N'" >> /tmp/commands.sql
    echo ".timeout 30000" >> /tmp/commands.sql
    echo ".output $4" >> /tmp/commands.sql
    echo "select * from $2 where dateTime > '$3' order by dateTime ASC;" >> /tmp/commands.sql
    echo ".exit" >> /tmp/commands.sql

    echo ".read /tmp/commands.sql" | sqlite3 $1

    rm -rf /tmp/commands.sql
}

# No function args
get_sql_server_config()
{
    echo "select value from config where name='STATION_SQLDB_HOST';" > /tmp/commands.sql
    echo ".exit" >> /tmp/commands.sql
    echo ".read /tmp/commands.sql" | sqlite3 $WVIEW_CONF_DIR/wview-conf.sdb > /tmp/result.txt
    SQL_HOST=`cat /tmp/result.txt`
    rm -rf /tmp/commands.sql /tmp/result.txt

    echo "select value from config where name='STATION_SQLDB_USERNAME';" > /tmp/commands.sql
    echo ".exit" >> /tmp/commands.sql
    echo ".read /tmp/commands.sql" | sqlite3 $WVIEW_CONF_DIR/wview-conf.sdb > /tmp/result.txt
    SQL_USERNAME=`cat /tmp/result.txt`
    rm -rf /tmp/commands.sql /tmp/result.txt

    echo "select value from config where name='STATION_SQLDB_PASSWORD';" > /tmp/commands.sql
    echo ".exit" >> /tmp/commands.sql
    echo ".read /tmp/commands.sql" | sqlite3 $WVIEW_CONF_DIR/wview-conf.sdb > /tmp/result.txt
    SQL_PASSWORD=`cat /tmp/result.txt`
    rm -rf /tmp/commands.sql /tmp/result.txt

    echo "select value from config where name='STATION_SQLDB_DB_NAME';" > /tmp/commands.sql
    echo ".exit" >> /tmp/commands.sql
    echo ".read /tmp/commands.sql" | sqlite3 $WVIEW_CONF_DIR/wview-conf.sdb > /tmp/result.txt
    SQL_DB_NAME=`cat /tmp/result.txt`
    rm -rf /tmp/commands.sql /tmp/result.txt
}

# $1 = import filename
mysql_import()
{
    if [ "$VERBOSE" = "1" ]; then
        date >> $MYSQL_IMPORT_LOG_FILE;
        mysqlimport --local --compress --replace --fields-terminated-by=',' \
                    --host=$SQL_HOST --user=$SQL_USERNAME --password=$SQL_PASSWORD \
                    $SQL_DB_NAME $1 >> $MYSQL_IMPORT_LOG_FILE;
    else
        mysqlimport --local --compress --replace --fields-terminated-by=',' \
                    --host=$SQL_HOST --user=$SQL_USERNAME --password=$SQL_PASSWORD \
                    $SQL_DB_NAME $1 > /dev/null;
    fi
}

################################################################################
##################  S C R I P T  E X E C U T I O N  S T A R T  #################
################################################################################

## Are we creating the export database?
if [ "$DO_CREATION" = "1" ]; then
    get_sql_server_config
    if [ x"$SQL_USERNAME" = x ]; then
        syslog_error "Failed to retrieve MySQL export parameters from $WVIEW_CONF_DIR/wview-conf.sdb"
        exit 1
    else
        syslog "Creating with: $SQL_USERNAME:<password omitted>:$SQL_DB_NAME"
    fi

    wview-mysql-create root $MYSQL_ROOT_PASSWORD $SQL_USERNAME $SQL_PASSWORD $SQL_DB_NAME
    exit 0;    
fi


UPDATE_TIME_SAVE=`date +%s`

## Check to see if there is new data to export:
if [ ! -e $TIME_MARKER_FILE ]; then
    LAST_UPDATE=0
    syslog "First time to export databases"
else
    LAST_UPDATE=`cat $TIME_MARKER_FILE`
    syslog "Exporting databases after: " $LAST_UPDATE
fi

if [ -e $ARCHIVE_MARKER_FILE ]; then
    if [ $LAST_UPDATE -ge `cat $ARCHIVE_MARKER_FILE` ]; then
        ARCHIVE_ENABLE=0
    fi
else
    syslog $ARCHIVE_MARKER_FILE " not found"
fi
if [ -e $HILOW_MARKER_FILE ]; then
    if [ $LAST_UPDATE -ge `cat $HILOW_MARKER_FILE` ]; then
        HILOW_ENABLE=0
    fi
else
    syslog $HILOW_MARKER_FILE " not found"
fi
if [ -e $NOAA_MARKER_FILE ]; then
    if [ $LAST_UPDATE -ge `cat $NOAA_MARKER_FILE` ]; then
        NOAA_ENABLE=0
    fi
else
    syslog $NOAA_MARKER_FILE " not found"
fi


## Export from the wview SQLite3 databases:
if [ "$ARCHIVE_ENABLE" = "1" ]; then
    UPDATE_MARKER=1
    syslog "Exporting to: " $ARCHIVE_EXPORT_FILE
    rm -rf $ARCHIVE_EXPORT_FILE
    sqlite3_export $ARCHIVE_DB_FILE $SQL_ARCHIVE_TABLE $LAST_UPDATE $ARCHIVE_EXPORT_FILE
fi

SAVE_UPDATE_TIME=$LAST_UPDATE
LAST_UPDATE=`expr $LAST_UPDATE - $SECONDS_IN_HOUR`

if [ "$HILOW_ENABLE" = "1" ]; then
    UPDATE_MARKER=1
    syslog "Exporting HILOW tables:"
    rm -rf $HILOW_EXPORT_FILE_INTEMP
    sqlite3_export $HILOW_DB_FILE $SQL_HILOW_TABLE_INTEMP $LAST_UPDATE $HILOW_EXPORT_FILE_INTEMP
    rm -rf $HILOW_EXPORT_FILE_OUTTEMP
    sqlite3_export $HILOW_DB_FILE $SQL_HILOW_TABLE_OUTTEMP $LAST_UPDATE $HILOW_EXPORT_FILE_OUTTEMP
    rm -rf $HILOW_EXPORT_FILE_INHUMIDITY
    sqlite3_export $HILOW_DB_FILE $SQL_HILOW_TABLE_INHUMIDITY $LAST_UPDATE $HILOW_EXPORT_FILE_INHUMIDITY
    rm -rf $HILOW_EXPORT_FILE_OUTHUMIDITY
    sqlite3_export $HILOW_DB_FILE $SQL_HILOW_TABLE_OUTHUMIDITY $LAST_UPDATE $HILOW_EXPORT_FILE_OUTHUMIDITY
    rm -rf $HILOW_EXPORT_FILE_BP
    sqlite3_export $HILOW_DB_FILE $SQL_HILOW_TABLE_BP $LAST_UPDATE $HILOW_EXPORT_FILE_BP
    rm -rf $HILOW_EXPORT_FILE_WINDSPEED
    sqlite3_export $HILOW_DB_FILE $SQL_HILOW_TABLE_WINDSPEED $LAST_UPDATE $HILOW_EXPORT_FILE_WINDSPEED
    rm -rf $HILOW_EXPORT_FILE_WINDGUST
    sqlite3_export $HILOW_DB_FILE $SQL_HILOW_TABLE_WINDGUST $LAST_UPDATE $HILOW_EXPORT_FILE_WINDGUST
    rm -rf $HILOW_EXPORT_FILE_DEWPOINT
    sqlite3_export $HILOW_DB_FILE $SQL_HILOW_TABLE_DEWPOINT $LAST_UPDATE $HILOW_EXPORT_FILE_DEWPOINT
    rm -rf $HILOW_EXPORT_FILE_RAIN
    sqlite3_export $HILOW_DB_FILE $SQL_HILOW_TABLE_RAIN $LAST_UPDATE $HILOW_EXPORT_FILE_RAIN
    rm -rf $HILOW_EXPORT_FILE_RAINRATE
    sqlite3_export $HILOW_DB_FILE $SQL_HILOW_TABLE_RAINRATE $LAST_UPDATE $HILOW_EXPORT_FILE_RAINRATE
    rm -rf $HILOW_EXPORT_FILE_WINDCHILL
    sqlite3_export $HILOW_DB_FILE $SQL_HILOW_TABLE_WINDCHILL $LAST_UPDATE $HILOW_EXPORT_FILE_WINDCHILL
    rm -rf $HILOW_EXPORT_FILE_HEATINDEX
    sqlite3_export $HILOW_DB_FILE $SQL_HILOW_TABLE_HEATINDEX $LAST_UPDATE $HILOW_EXPORT_FILE_HEATINDEX
    rm -rf $HILOW_EXPORT_FILE_ET
    sqlite3_export $HILOW_DB_FILE $SQL_HILOW_TABLE_ET $LAST_UPDATE $HILOW_EXPORT_FILE_ET
    rm -rf $HILOW_EXPORT_FILE_UV
    sqlite3_export $HILOW_DB_FILE $SQL_HILOW_TABLE_UV $LAST_UPDATE $HILOW_EXPORT_FILE_UV
    rm -rf $HILOW_EXPORT_FILE_SOLARRAD
    sqlite3_export $HILOW_DB_FILE $SQL_HILOW_TABLE_SOLARRAD $LAST_UPDATE $HILOW_EXPORT_FILE_SOLARRAD
    rm -rf $HILOW_EXPORT_FILE_HAIL
    sqlite3_export $HILOW_DB_FILE $SQL_HILOW_TABLE_HAIL $LAST_UPDATE $HILOW_EXPORT_FILE_HAIL
    rm -rf $HILOW_EXPORT_FILE_HAILRATE
    sqlite3_export $HILOW_DB_FILE $SQL_HILOW_TABLE_HAILRATE $LAST_UPDATE $HILOW_EXPORT_FILE_HAILRATE
    rm -rf $HILOW_EXPORT_FILE_WINDDIR
    sqlite3_export $HILOW_DB_FILE $SQL_HILOW_TABLE_WINDDIR $LAST_UPDATE $HILOW_EXPORT_FILE_WINDDIR
fi

LAST_UPDATE=`expr $SAVE_UPDATE_TIME - $SECONDS_IN_DAY`

if [ "$NOAA_ENABLE" = "1" ]; then
    UPDATE_MARKER=1
    syslog "Exporting to: " $NOAA_EXPORT_FILE
    rm -rf $NOAA_EXPORT_FILE
    sqlite3_export $NOAA_DB_FILE $SQL_NOAA_TABLE $LAST_UPDATE $NOAA_EXPORT_FILE
fi


## Get MySQL server configuration:
get_sql_server_config
if [ x"$SQL_USERNAME" = x ]; then
    syslog_error "Failed to retrieve MySQL export parameters from $WVIEW_CONF_DIR/wview-conf.sdb"
    exit 1
else
    syslog "Exporting with: $SQL_HOST:$SQL_USERNAME:<password omitted>:$SQL_DB_NAME"
fi


## Import to MySQL:
if [ "$ARCHIVE_ENABLE" = "1" ]; then
    syslog "Importing: " $ARCHIVE_EXPORT_FILE
    mysql_import $ARCHIVE_EXPORT_FILE
fi

if [ "$HILOW_ENABLE" = "1" ]; then
    syslog "Importing HILOW tables:"
    mysql_import $HILOW_EXPORT_FILE_INTEMP
    mysql_import $HILOW_EXPORT_FILE_OUTTEMP
    mysql_import $HILOW_EXPORT_FILE_INHUMIDITY
    mysql_import $HILOW_EXPORT_FILE_OUTHUMIDITY
    mysql_import $HILOW_EXPORT_FILE_BP
    mysql_import $HILOW_EXPORT_FILE_WINDSPEED
    mysql_import $HILOW_EXPORT_FILE_WINDGUST
    mysql_import $HILOW_EXPORT_FILE_DEWPOINT
    mysql_import $HILOW_EXPORT_FILE_RAIN
    mysql_import $HILOW_EXPORT_FILE_RAINRATE
    mysql_import $HILOW_EXPORT_FILE_WINDCHILL
    mysql_import $HILOW_EXPORT_FILE_HEATINDEX
    mysql_import $HILOW_EXPORT_FILE_ET
    mysql_import $HILOW_EXPORT_FILE_UV
    mysql_import $HILOW_EXPORT_FILE_SOLARRAD
    mysql_import $HILOW_EXPORT_FILE_HAIL
    mysql_import $HILOW_EXPORT_FILE_HAILRATE
    mysql_import $HILOW_EXPORT_FILE_WINDDIR
fi

if [ "$NOAA_ENABLE" = "1" ]; then
    syslog "Importing: " $NOAA_EXPORT_FILE
    mysql_import $NOAA_EXPORT_FILE
fi


## Save the current time marker:
if [ "$UPDATE_MARKER" = "1" ]; then
    echo $UPDATE_TIME_SAVE > $TIME_MARKER_FILE
fi

## Done!
exit 0

